### ISSUE TITLE
cannot compile new angular projecy

### ISSUE THREAD

**Issue by lucamorelli** [aspnet/AspNetCore#17919]
### Describe the bug
I created a new angular project using the latest versions of dotnet new and angular template, and if I try to run via ng serve obtain these results:

error TS2345: Argument of type 'Profile' is not assignable to parameter of type 'IUser'.
  Property 'name' is optional in type 'Profile' but required in type 'IUser'.

77       this.userSubject.next(user.profile);
                               ~~~~~~~~~~~~
api-authorization/authorize.service.ts:88:31 - error TS2345: Argument of type 'Profile' is not assignable to parameter of type 'IUser'.

88         this.userSubject.next(user.profile);
                                 ~~~~~~~~~~~~
api-authorization/authorize.service.ts:114:29 - error TS2345: Argument of type 'Profile' is not assignable to parameter of type 'IUser'.

114       this.userSubject.next(user && user.profile);
                                ~~~~~~~~~~~~~~~~~~~~
api-authorization/authorize.service.ts:149:42 - error TS2339: Property 'data' does not exist on type 'SignoutResponse'.

149       return this.success(state && state.data);
                                             ~~~~
api-authorization/authorize.service.ts:194:5 - error TS2322: Type 'Observable<Profile>' is not assignable to type 'Observable<IUser>'.
  Type 'Profile' is not assignable to type 'IUser'.

194     return from(this.ensureUserManagerInitialized())
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
195       .pipe(
    ~~~~~~~~~~~~
196         mergeMap(() => this.userManager.getUser()),
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
197         map(u => u && u.profile));
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

### To Reproduce
create a new project suing the dot net new angular command and try to run it

### Further technical details
.NET Core SDK (che rispecchia un qualsiasi file global.json):
 Version:   3.1.100
 Commit:    cd82f021f4

Ambiente di runtime:
 OS Name:     Windows
 OS Version:  10.0.18362
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\3.1.100\

Host (useful for support):
  Version: 3.1.0
  Commit:  65f04fb6db

.NET Core SDKs installed:
  1.1.14 [C:\Program Files\dotnet\sdk]
  2.2.207 [C:\Program Files\dotnet\sdk]
  3.1.100 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.14 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.14 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.16 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.13 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.14 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.2.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.0 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

indipendent from the IDE

**Comment by javiercn** [comment]
@lucamorelli thanks for contacting us.

I'm not able to repro this on an isolated environment. There might be something on your environment affecting the typescript compilation. I would suggest you make sure you have an update node/npm version and potentially clean your npm package cache if that doesn't solve it.

**Comment by SyntaxWarrior** [comment]
I also ran into this. I have a project that I have been working on for a while and got this after trying to get rid of this error: 

`browserlist: caniuse-lite is outdated`

I followed this advice: https://stackoverflow.com/a/58523903/937131
And then did an npm update / install

This also seems to have moved "oidc-client": to a higer version "1.10.1", not sure if this is related. 

And now I have this same problem as described above.

**Comment by s-whaley** [comment]
I ran into this issue as well. I believe it is related to the updated oidc-client package 1.10.x release which added a new type for 'Profile'. I do not think this is related to ASpNetCore.

**Comment by SyntaxWarrior** [comment]
How did you resolve the issue?
I just downgraded to old oidc client and still get the error.

**Comment by benday** [comment]
@SyntaxWarrior  -- i ran into the same problem.  I changed it in my package.json but that wasn't enough because it was still being referenced (i assume) through package-lock.json.  

The command I ran that fixed it was 
`npm i oidc-client@1.9.1`

The only problem is that it gave me a whole new giant pile of errors related to 'routerLink'.  It looks like there are a handful of places where RouterTestingModule needs to be imported into the api-authorization specs.
